CodePushPackage.m

```diff
@@ -285,6 +285,11 @@ + (void)downloadPackage:(NSDictionary *)updatePackage
                 [CodePushPackage copyEntriesInFolder:unzippedFolderPath
                                           destFolder:newPackageFolderPath
                                                error:&error];
+                if (error) {
```

Comment:
1. This is the same behaviour as Android https://github.com/Microsoft/react-native-code-push/blob/master/android/app/src/main/java/com/microsoft/codepush/react/CodePush.java#L379: If the error is an IOException (which should not happen), we don't mark the update failed because the problem is with our plugin, not with the update. I don't want to overengineer the thing by adding retry logic. I think the simplest solution is to reject the sync promise, and get the error printed to the console so that users can report it. I think that if this is in place, we would at least get more information on https://github.com/Microsoft/react-native-code-push/issues/180

2. Good point, and totally agree, we should delete it before returning.